### PR TITLE
feat: credentials payload verification

### PR DESCRIPTION
**Pull request by michaelavoyan**
This PR introduces a complete implementation of credential payload verifiers for validating JWT-based Verifiable Credentials in compliance with the Velocity Profile and partially with OpenID4VCI specification.

🔍 What’s Implemented
	•	Modular, functional verifiers for:
	•	alg (JWT header algorithm)
	•	iss (issuer) – both strict and spec-fallback modes
	•	kid (key ID prefix)
	•	sub or cnf (subject binding)
	•	vc.credentialSchema presence
	•	vc.credentialStatus presence
	•	verifyCredentialJwtPayloadStrict() – composition of verifiers for profile-level validation
	•	verifyCredentialEndpointResponse() – validates array of credentials from Credential Endpoint responses
	•	Extensive unit test coverage for all rules and edge cases
	•	Strong TS documentation blocks for maintainability and clarity

❗Note:
All verifiers were implemented **manually**, without relying on `credo-ts`.

🚧 Next Steps
	•	Investigate integration options with `credo-ts` verifiers
	•	Add the `credo-ts` verifiers under `impl/verifiers/credo-ts-verifiers`

### CODE REVIEW

File: packages/velocity-exchange-verifiers/src/impl/rules/verify-credential-jwt-payload-strict.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * }
+ *
+ * @param credential - A parsed Credential JWT containing `header` and `payload`
+ * @param context - Validation context including issuer metadata and optional path
+ * @returns An array of `VerificationError` objects, or an empty array if the credential is valid
+ *
+ * @see {@link CredentialJwt}
+ * @see {@link VerificationError}
+ */
+export const verifyCredentialJwtPayloadStrict = createVerifier<any>([
```
[the remark this answers is not in the document]
Comment (sloops77) [reply packages/velocity-exchange-verifiers/src/impl/rules/verify-credential-jwt-payload-strict.ts:55]: and in general. dont use the `any` type

**Review by sloops77 — CHANGES_REQUESTED** [review]
In general it looks good

File: packages/velocity-exchange-verifiers/src/impl/verifiers/pure-verifiers/credential-status-verifier.ts [hunk 2]
```diff
@@ -33,17 +33,19 @@ import { CredentialJwt, Verifier } from 'impl/types';
...
  */
-export const credentialStatusVerifier: Verifier<CredentialJwt> = (
+export const credentialStatusVerifier: Verifier<W3CCredentialJwtV1> = (
   credential,
   context
 ) => {
   const path = [...(context.path ?? []), 'payload', 'vc', 'credentialStatus'];
 
-  if (!_.has(credential.payload, 'vc.credentialStatus')) {
+  const exists = credential.payload?.vc?.credentialStatus != null;
```
Comment (sloops77) [inline packages/velocity-exchange-verifiers/src/impl/verifiers/pure-verifiers/credential-status-verifier.ts:46]: rename `exists` to `credentialStatusExists`

File: packages/velocity-exchange-verifiers/src/impl/verifiers/pure-verifiers/iss-claim-matches-either-metadata-or-credential-issuer-verifier.ts [hunk 3]
```diff
@@ -35,12 +35,12 @@ import { buildError, ERROR_CODES } from 'impl/errors';
...
+ * @see {@link W3CCredentialJwtV1}
  * @see {@link VerificationError}
  * @see {@link VerificationContext}
  */
 export const issClaimMatchesEitherMetadataOrCredentialIssuerVerifier: Verifier<
-  CredentialJwt
+  W3CCredentialJwtV1
 > = (credential, context) => {
   const actual = credential.payload?.iss;
   const allowed = [
```
Comment (sloops77) [inline packages/velocity-exchange-verifiers/src/impl/verifiers/pure-verifiers/iss-claim-matches-either-metadata-or-credential-issuer-verifier.ts:46]: - this code should assume that context.credential_issuer_metadata is NOT NULL
- If `iss` or `credential_issuer` is null, then it IS NOT an `allowed` value so should be removed. 
- also `payload` cannot be null, nor can `payload.iss`

File: packages/velocity-exchange-verifiers/src/impl/verifiers/pure-verifiers/alg-is-supported-verifier.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+ * @see {@link CredentialJwt}
+ * @see {@link VerificationError}
+ */
+export const algIsSupportedVerifier: Verifier<CredentialJwt> = (
+  credential,
+  context
+) => {
+  const alg = credential.header?.alg;
+  if (!supportedAlgs.includes(alg)) {
+    return [
```
[the remark this answers is not in the document]
Comment (sloops77) [reply packages/velocity-exchange-verifiers/src/impl/verifiers/pure-verifiers/alg-is-supported-verifier.ts:47]: not resolved

File: packages/velocity-exchange-verifiers/src/impl/verifiers/pure-verifiers/iss-claim-matches-either-metadata-or-credential-issuer-verifier.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+export const issClaimMatchesEitherMetadataOrCredentialIssuerVerifier: Verifier<
+  CredentialJwt
+> = (credential, context) => {
+  const actual = credential.payload?.iss;
+  const allowed = [
+    context.credential_issuer_metadata?.iss,
+    context.credential_issuer_metadata?.credential_issuer,
+  ];
+
+  if (!actual || !allowed.includes(actual)) {
```
[the remark this answers is not in the document]
Comment (sloops77) [reply packages/velocity-exchange-verifiers/src/impl/verifiers/pure-verifiers/iss-claim-matches-either-metadata-or-credential-issuer-verifier.ts:51]: not resolved

this code should assume that `context.credential_issuer_metadata` is NOT NULL
If `iss` or `credential_issuer` is `null`, then it IS NOT `allowed` and should be removed from the array.
also `payload` cannot be `null`, nor can `payload.iss`

**Review by sloops77 — CHANGES_REQUESTED** [review]

File: packages/velocity-exchange-verifiers/src/impl/verifiers/pure-verifiers/alg-is-supported-verifier.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (michaelavoyan) [reply packages/velocity-exchange-verifiers/src/impl/verifiers/pure-verifiers/alg-is-supported-verifier.ts:47]: @sloops77 I think my current implementation already meets the "one rule - one error" requirement. Can you please elaborate?
